test/ibmq/test_ibmq_job.py

```diff
@@ -289,12 +291,109 @@ def test_retrieve_job_error(self, provider):
+            # Try twice in case job is not in a cancellable state
+            try:
+                if job_to_cancel.cancel():
+                    status = job_to_cancel.status()
+                    # TODO Change the warning to assert once API is fixed
+                    if status is not JobStatus.CANCELLED:
+                        self.log.warning("cancel() was successful for job %s but its status is %s.",
+                                         job_to_cancel.job_id(), status)
+            except JobError:
+                pass
+
+        # Submit a job that will fail.
+        qobj.config.shots = 10000  # Modify the number of shots to be an invalid amount.
+        job_to_fail = backend.run(qobj)
+        while job_to_fail.status() not in JOB_FINAL_STATES:
+            time.sleep(0.5)
+
+        for status_filter in status_filters:
+            with self.subTest(status_filter=status_filter):
+                job_list = backend.jobs(status=status_filter['status'],
+                                        db_filter=status_filter['db_filter'])
+
+                if (job_to_cancel.status() is JobStatus.CANCELLED
+                        and job_to_fail.status() is JobStatus.ERROR):
+                    job_list_ids = [_job.job_id() for _job in job_list]
+                    # Assert `job_id` in the list of job id's (instead of the list of jobs),
+                    # because retrieved jobs might differ in attributes from the originally
+                    # submitted jobs.
+                    self.assertIn(job_to_fail.job_id(), job_list_ids)
+                    self.assertIn(job_to_cancel.job_id(), job_list_ids)
+
+                for filtered_job in job_list:
+                    self.assertIn(filtered_job._status, statuses_to_filter,
+                                  "job {} has status '{}' but it should be one "
+                                  "of the values '{}'"
+                                  .format(filtered_job.job_id(), filtered_job._status,
+                                          statuses_to_filter))
+
+    @requires_provider
+    def test_retrieve_active_jobs(self, provider):
+        """Test retrieving jobs that are currently unfinished."""
+        backend = max([b for b in provider.backends(simulator=False) if b.status().operational],
+                      key=lambda b: b.status().pending_jobs)
+        active_job_statuses = {api_status_to_job_status(status) for status in ApiJobStatus
+                               if status not in API_JOB_FINAL_STATES}
+
+        qobj = assemble(transpile(self._qc, backend=backend), backend=backend)
+        job = backend.run(qobj)
+        # Wait for the job to queue, run, or finish.
+        while job.status() not in list(JOB_FINAL_STATES) + [JobStatus.QUEUED, JobStatus.RUNNING]:
+            time.sleep(0.5)
+
+        before_status = job.status()
+        active_jobs = backend.active_jobs()
+        if before_status is JobStatus.QUEUED and job.status() is JobStatus.QUEUED:
```

Comment:
Similar to above, the job doesn't have to be QUEUED, it just needs not be in final state. There's also no need to check for before status.